### PR TITLE
Challenge Subnav: Priority Nav Refactor

### PR DESCRIPTION
**Pull request by nemaniarjun**

### CODE REVIEW

File: app/assets/javascripts/controllers/challenges_controller.js [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+    init: function() {
+        this.menu = $(".priority-nav");
+        this.allNavElements = $(".priority-nav > ul > li:not('.overflow-nav')");
+        this.bindUIActions();
+        this.setupMenu();
+    },
+
+    setupMenu: function() {
+        let firstPos = $(".priority-nav > ul > li:first").position();
+        let wrappedElements = $();
```
Comment (piotrekpasciak) [inline app/assets/javascripts/controllers/challenges_controller.js]: Are you sure this line will work? Can you do something like `$()`? What will it do?

File: app/helpers/challenges_helper.rb [hunk 2]
```diff
@@ -12,27 +12,17 @@ def status_dropdown_helper(builder:)
...
     if !policy(challenge).has_accepted_participation_terms?
-      return url_for([challenge, ParticipationTerms.current_terms])
+      [challenge, ParticipationTerms.current_terms]
     elsif !policy(challenge).has_accepted_challenge_rules?
-      return url_for([challenge, challenge.current_challenge_rules])
+      [challenge, challenge.current_challenge_rules]
     end
 
-    return nil
+    nil
```
Comment (piotrekpasciak) [inline app/helpers/challenges_helper.rb]: `nil` here is redundant. Methods in Ruby return nil by default.

**Comment by piotrekpasciak** [comment]
Looks like some specs don't pass, please take a look on them.

File: app/assets/javascripts/controllers/challenges_controller.js (hunk 1 above)
Comment (nemaniarjun) [reply app/assets/javascripts/controllers/challenges_controller.js]: It creates an empty set / array for jquery to add elements too

**Comment by nemaniarjun** [comment]
> Looks like some specs don't pass, please take a look on them.

will fix the specs today!

**Pull request merged**